tests/spl/token-extensions/programs/token-extensions/src/instructions.rs

```diff
@@ -0,0 +1,169 @@
+use anchor_lang::{prelude::*, solana_program::entrypoint::ProgramResult};
+
+use anchor_spl::{
+    associated_token::AssociatedToken,
+    token_2022::spl_token_2022::extension::{
+        group_member_pointer::GroupMemberPointer, metadata_pointer::MetadataPointer,
+        mint_close_authority::MintCloseAuthority, permanent_delegate::PermanentDelegate,
+        transfer_hook::TransferHook,
+    },
+    token_interface::{
+        spl_token_2022::extension::ExtensionType,
+        spl_token_metadata_interface::state::TokenMetadata, token_metadata_initialize, Mint,
+        Token2022, TokenAccount, TokenMetadataInitialize, TokenMetadataInitializeArgs,
+    },
+};
+use spl_pod::optional_keys::OptionalNonZeroPubkey;
+
+use crate::{
+    get_meta_list_size, get_mint_extensible_extension_data, get_mint_extension_data,
+    update_account_lamports_to_minimum_balance, META_LIST_ACCOUNT_SEED,
+};
+
+#[derive(AnchorDeserialize, AnchorSerialize)]
+pub struct CreateMintAccountArgs {
+    pub name: String,
+    pub symbol: String,
+    pub uri: String,
+}
+
+pub const MINT_EXTENSIONS: [ExtensionType; 6] = [
+    ExtensionType::MetadataPointer,
+    ExtensionType::GroupMemberPointer,
+    ExtensionType::TransferHook,
+    ExtensionType::MintCloseAuthority,
+    ExtensionType::PermanentDelegate,
+    ExtensionType::NonTransferable,
+];
+
+#[derive(Accounts)]
+#[instruction(args: CreateMintAccountArgs)]
+pub struct CreateMintAccount<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account(mut)]
+    /// CHECK: can be any account
+    pub authority: Signer<'info>,
+    #[account()]
+    /// CHECK: can be any account
+    pub receiver: UncheckedAccount<'info>,
+    #[account(
+        init,
+        signer,
+        payer = payer,
+        mint::token_program = token_program,
+        mint::decimals = 0,
+        mint::authority = authority,
+        mint::freeze_authority = authority,
+        mint::extensions = MINT_EXTENSIONS.to_vec(),
+        extensions::metadata_pointer::authority = authority.key(),
+        extensions::metadata_pointer::metadata_address = mint.key(),
+        extensions::group_member_pointer::authority = authority.key(),
+        extensions::group_member_pointer::member_address = mint.key(),
+        extensions::transfer_hook::authority = authority.key(),
+        extensions::transfer_hook::program_id = crate::ID,
+        extensions::close_authority::authority = authority.key(),
+        extensions::permanent_delegate::delegate = authority.key(),
```

Comment:
I still think it would be better to keep things consistent and have implicit `key()` call on at least most of these constraints, since most of them need to be validated, especially when it's not `init`.

Reply:
I think you're right, I will update to use .key()